lib/eyaml/encryption_manager.rb

```diff
@@ -0,0 +1,96 @@
+class EYAML
+  class EncryptionManager
+    FORMAT_REGEX = /\AEJ\[(?<version>[^:]+):(?<session_public_key>[^:]+):(?<nonce>[^:]+):(?<text>[^\]]+)\]\z/
+    FORMAT_VERSION = "1"
+
+    class << self
+      def generate_key
+        private_key = RbNaCl::PrivateKey.generate
+
+        [
+          RbNaCl::Util.bin2hex(private_key.public_key),
+          RbNaCl::Util.bin2hex(private_key)
+        ]
+      end
+    end
+
+    def initialize(yaml, public_key, private_key)
+      @tree = yaml
+      @public_key = public_key.encoding != Encoding::BINARY ? RbNaCl::Util.hex2bin(public_key)
+                                                            : public_key
+      @private_key = private_key.encoding != Encoding::BINARY ? RbNaCl::Util.hex2bin(private_key)
+                                                              : private_key
```

Comment:
You should create a helper for this logic

Reply:
Good call 👌